### PR TITLE
[DM-347] As user I want user docs for advanced software management

### PR DESCRIPTION
**Pull request by bozhidardedov**

### CODE REVIEW

**Review by rssag — APPROVED** [review]
language review as commit + approval

File: content/users-guide/device-management-bundle/device-details.md [hunk 1]
```diff
@@ -393,15 +393,19 @@ To configure LAN parameters, simply enter **IP address** and **Subnet mask**.
...
 ![Services overview](/images/users-guide/DeviceManagement/devmgmt-services-overview.png)
 
-The platform defines three status options for services - **Up**, **Down** and **Unknown**. These pre-defined status have their own graphical represenation and can be directly selected from **Status** filter:
+The platform defines three status options for services: **Up**, **Down** and **Unknown**.
+These predefined statuses have their own graphical represenation and can be selected directly from the **Status** filter:
 
 ![Status filter](/images/users-guide/DeviceManagement/devmgmt-services-status-filter.png)
 
-Other status options are also allowed and can be freely defined. They all would share the same icon and need to be manually entered in the **Status** filter. A tooltip on the status icon will display their actual value:
+Other status options are also allowed and can be freely defined.
```
Comment (dafkk) [inline content/users-guide/device-management-bundle/device-details.md:406]: I was wondering if we can add here a link to the resources docs with the way to add a new option, but I am not sure if that is published. I am talking about this here - packages/ngx-components/services/services.module.md. Does anyone knows @rssag, @BeateRixen?

**Review by dafkk — APPROVED** [review]

File: content/users-guide/device-management-bundle/managing-device-data.md [hunk 2]
```diff
@@ -189,14 +189,17 @@ The available software objects will be displayed as a list.
...
 
 When clicking on an entry, the details for this software are displayed along with all available versions.
 
 ![Software details](/images/users-guide/DeviceManagement/devmgmt-software-details.png)
 
-At the top, the software name, a description, and optional device type filter(s) are shown. If a filter is set, the software will show up for installation only for devices of that type. If no filter is set, it will be available for all devices.
+At the top, the software name, a description, an optional device type filter(s), and a software type are shown.
+If a device type filter is set, the software will show up for installation only for devices of that type.
+If no filter is set, it will be available for all devices.
+The software type will make the software installable only on devices that have declared to support the particular software type.
```
Comment (BeateRixen) [inline content/users-guide/device-management-bundle/managing-device-data.md:202]: @rssag "only on devices that have declared to support" 
I think we can get that simpler :-). Suggestion:  "only on devices that explicitly support"? Also a bit further below.

File: content/users-guide/device-management-bundle/managing-device-data.md [hunk 3]
```diff
@@ -260,9 +264,19 @@ Additionally, it shows the operation status for the last operation (one of SUCCE
...
-2. Select one or multiple software items by selecting the respective version from the list, which contains all software for the particular device type available in the software repository.
-4. Click **Install**.
-5. In the **Software changes** panel at the right, review your planned changes and confirm the software update operation by clicking **Apply changes**.<br><br>
+
+    >**Info:** The **Install software** dialog will only display software entries matching the device type. Additionally, if the device has any `c8y_SupportedSoftwareTypes` declared the dialog will only display the software entries matching the supported software types.
+
+2. Select one or multiple softwares by selecting the respective version from the list which contains all softwares for the particular device type available in the software repository. For devices supporting advanced software management features, already installed software entries cannot be pre-filtered from the list of available software entries. Thus, after a particular software version has been selected, a check is done if the chosen software is already installed on the device. If this is the case a warning is displayed:
```
Comment (BeateRixen) [inline content/users-guide/device-management-bundle/managing-device-data.md:270]: softwares > software items; several times
software entries > software item
chosen > selected

File: content/users-guide/device-management-bundle/managing-device-data.md [hunk 4]
```diff
@@ -260,9 +264,19 @@ Additionally, it shows the operation status for the last operation (one of SUCCE
...
+
+    ![Already installed software entry warning](/images/users-guide/DeviceManagement/devmgmt-software-install-warning.png)
+
+    You can remove the already installed software entry from the **Software changes** panel. However, if the resulting operation includes an already installed software entry this has no effect on the device software.
```
Comment (BeateRixen) [inline content/users-guide/device-management-bundle/managing-device-data.md:274]: entry > item
panel > we don´t use this; not sure what this looks like?

File: content/users-guide/device-management-bundle/managing-device-data.md (hunk 2 above)
Comment (rssag) [reply content/users-guide/device-management-bundle/managing-device-data.md:202]: i like "that specifically support"

File: content/users-guide/device-management-bundle/managing-device-data.md [hunk 5]
```diff
@@ -260,9 +264,19 @@ Additionally, it shows the operation status for the last operation (one of SUCCE
...
+
+    ![Software changes warning](/images/users-guide/DeviceManagement/devmgmt-software-changes-warning.png)
+
+3. Click **Install**.
+4. In the **Software changes** panel at the right, review your planned changes and confirm the software update operation by clicking **Apply changes**.<br><br>
```
Comment (BeateRixen) [inline content/users-guide/device-management-bundle/managing-device-data.md:279]: panel?

File: content/users-guide/device-management-bundle/device-details.md (hunk 1 above)
Comment (rssag) [reply content/users-guide/device-management-bundle/device-details.md:406]: what would the full link be?

**Comment by BeateRixen** [comment]
@bozhidardedov There are a couple of screenshots showing tooltips or messages. I would avoid these kinds of screenshots and remove them. They are of not much value for the user. Instead, you could simply provide the warning/tooltip message in the text. Be aware that screenshot maintenance kills us (for example, if we again have a design change) :-).

File: content/users-guide/device-management-bundle/device-details.md (hunk 1 above)
Comment (bozhidardedov) [reply content/users-guide/device-management-bundle/device-details.md:406]: The link to this part of the published WebSDK documentation should be http://resources.cumulocity.com/documentation/websdk/ngx-components/modules/ServicesModule.html#readme

image

File: content/users-guide/device-management-bundle/managing-device-data.md (hunk 2 above)
Comment (bozhidardedov) [reply content/users-guide/device-management-bundle/managing-device-data.md:202]: Device MOs can have a fragment in which the software types they support are listed. Thus "only on devices that explicitly support" should make that clear, I guess.

File: content/users-guide/device-management-bundle/managing-device-data.md (hunk 4 above)
Comment (bozhidardedov) [reply content/users-guide/device-management-bundle/managing-device-data.md:274]: Typo, "pane" is meant.

File: content/users-guide/device-management-bundle/managing-device-data.md (hunk 5 above)
Comment (bozhidardedov) [reply content/users-guide/device-management-bundle/managing-device-data.md:279]: Again "pane".

**Comment by bozhidardedov** [comment]
> @bozhidardedov There are a couple of screenshots showing tooltips or messages. I would avoid these kinds of screenshots and remove them. They are of not much value for the user. Instead, you could simply provide the warning/tooltip message in the text. Be aware that screenshot maintenance kills us (for example, if we again have a design change) :-).

Got that. I've removed those screenshots and changed text accordingly.

File: content/users-guide/device-management-bundle/managing-device-data.md (hunk 2 above)
Comment (rssag) [reply content/users-guide/device-management-bundle/managing-device-data.md:202]: done

File: content/users-guide/device-management-bundle/managing-device-data.md (hunk 3 above)
Comment (rssag) [reply content/users-guide/device-management-bundle/managing-device-data.md]: done, found a couple more occurrences.

File: content/users-guide/device-management-bundle/managing-device-data.md (hunk 4 above)
Comment (rssag) [reply content/users-guide/device-management-bundle/managing-device-data.md]: i think we can just avoid panel and only use the UI text.

File: content/users-guide/device-management-bundle/managing-device-data.md (hunk 5 above)
Comment (rssag) [reply content/users-guide/device-management-bundle/managing-device-data.md]: see above

File: content/users-guide/device-management-bundle/device-details.md (hunk 1 above)
Comment (rssag) [reply content/users-guide/device-management-bundle/device-details.md:406]: this yields a 404 - will the URL work at some point?

Comment (bozhidardedov) [reply content/users-guide/device-management-bundle/device-details.md:406]: Once WebSDK doc has been built for 10.14.0.x, the link should start working.

Comment (rssag) [reply content/users-guide/device-management-bundle/device-details.md]: added the URL

**Review by BeateRixen — APPROVED** [review]

**Pull request merged**